packages/jaeger-ui/src/types/trace.tsx

```diff
@@ -78,6 +78,12 @@ export type Trace = TraceData & {
   // Number of orphan spans (spans referencing parent spans that don't exist in the trace)
   orphanSpanCount?: number;
 
+  // Optimized data structures - created once during trace transformation
+  // Optional to support test scenarios where traces may be mocked
+  spanMap?: Map<string, Span>;
+  tree?: TreeNode<string>;
+  nodesBySpanId?: Map<string, TreeNode<string>>;
```

Comment:
make these fields required and remove the fallback initialization in production code. It doesn't make sense to pollute production call with fallbacks just because the tests are lazy - if necessary we can initialize the data better in the tests. Since these are computed fields anyway the tests are probably instantiating TraceData rather than post-processed Trace